### PR TITLE
Add structured output

### LINKED ISSUE #23: [Feature Request]: Structured Output
### Is your feature request related to a problem? Please describe.

Ive came across many comments from the community, asking for the formatted output/ structural output from the Agentic chat (initate_chat and groupchat ). I believe generating the expected response after the agentic chat can be the biggest improvement in AutoGen. 

Me, @marklysze and @sonichi discussed about adding this feature over a call. 

### Describe the solution you'd like

There should be an output message after the completion of the chat, i can see there is `last_message`, but that doesn't serve the exact feature. adding this feature can help production apps use AutoGen.

solution: TBD ( help and ideas are appreciated )

### Additional context

We can get started with the openai structured output and final goal should be the structured output for the complete chat.

cc @sonichi @BeibinLi @qingyun-wu @marklysze @yiranwu0 @skzhang1

### PR DESCRIPTION
**Pull request by davorrunje**
## Why are these changes needed?

Add `structured_format` parameter to the `ConversibleAgent` class and use it to define JSON output format of the agent.

## Related issue number

Closes #23 

## Checks

- [ ] I've included any doc changes needed for https://ag2ai.github.io/ag2/. See https://ag2ai.github.io/ag2/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by davorrunje** [comment]
We are waiting for @sternakt to push his changes to this branch.

**Comment by qingyun-wu** [comment]
> We are waiting for @sternakt to push his changes to this branch.

Thanks @davorrunje, changes are pushed in now? Let's convert this draft PR to a formal PR?

**Review by qingyun-wu — APPROVED** [review]
Looks good! Thanks @davorrunje!

File: notebook/agentchat_structured_outputs.ipynb [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+   "source": [
+    "user_proxy.initiate_chat(assistant, message=\"how can I solve 8x + 7 = -23\", max_turns=1, summary_method=\"last_msg\")"
+   ]
+  }
+ ],
+ "metadata": {
+  "front_matter": {
+   "description": "Explore the utilization of large language models in automated group chat scenarios, where agents perform tasks collectively, demonstrating how they can be configured, interact with each other, and retrieve specific information from external resources.",
+   "tags": [
+    "orchestration",
```
Comment (qingyun-wu) [inline notebook/agentchat_structured_outputs.ipynb:185]:
```suggestion
    "structured output",
```

File: notebook/agentchat_structured_outputs.ipynb [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    "group chat",
```
Comment (qingyun-wu) [inline notebook/agentchat_structured_outputs.ipynb:186]:
```suggestion
```

File: notebook/agentchat_structured_outputs.ipynb [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    "code generation"
```
Comment (qingyun-wu) [inline notebook/agentchat_structured_outputs.ipynb:187]:
```suggestion
```

File: notebook/agentchat_structured_outputs.ipynb [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    }
+   ],
+   "source": [
+    "user_proxy.initiate_chat(assistant, message=\"how can I solve 8x + 7 = -23\", max_turns=1, summary_method=\"last_msg\")"
+   ]
+  }
+ ],
+ "metadata": {
+  "front_matter": {
+   "description": "Explore the utilization of large language models in automated group chat scenarios, where agents perform tasks collectively, demonstrating how they can be configured, interact with each other, and retrieve specific information from external resources.",
```
Comment (qingyun-wu) [inline notebook/agentchat_structured_outputs.ipynb:183]:
```suggestion
   "description": "OpenAI offers a functionality for defining a structure of the messages generated by LLMs, AutoGen enables this functionality by propagating response_format passed to your agents to the underlying client.",
```

**Review by qingyun-wu — COMMENTED** [review]
Update description and tags of the notebook

**Comment by qingyun-wu** [comment]
@ohdearquant could you help review this? Please approve if it reads good to you! Thank you!

File: notebook/agentchat_structured_outputs.ipynb [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+   "source": [
+    "user_proxy.initiate_chat(assistant, message=\"how can I solve 8x + 7 = -23\", max_turns=1, summary_method=\"last_msg\")"
+   ]
+  }
+ ],
+ "metadata": {
+  "front_matter": {
+   "description": "OpenAI offers a functionality for defining a structure of the messages generated by LLMs, AutoGen enables this functionality by propagating response_format passed to your agents to the underlying client.",
+   "tags": [
+    "structured output",
```
Comment (qingyun-wu) [inline notebook/agentchat_structured_outputs.ipynb:185]:
```suggestion
    "structured output"
```

**Comment by qingyun-wu** [comment]
@davorrunje @sternakt It seems meta data of the notebook included in this PR was not updated properly. I made direct changes to the notebook. Please remember to check meta data in future notebooks. Thank you!

**Review by ohdearquant — APPROVED** [review]

**Review by ohdearquant — APPROVED** [review]
Looks good. Main changes are the response format and oai client, which is a clean straightforward implementation.

**Pull request merged**